src/core/meshcop/network_name.hpp

```diff
@@ -260,6 +260,14 @@ class NetworkNameManager : public InstanceLocator, private NonCopyable
      *
      */
     Error SetDomainName(const NameData &aNameData);
+
+    /**
+     * Checks whether the Thread Domain Name is currently set to the default name.
+     *
+     * @returns true if Thread Domain Name equals "DefaultDomain", false otherwise.
+     */
+    bool IsDefaultDomainNameSet();
```

Comment:
Also can consider declaring the method as `const`?
```suggestion
    bool IsDefaultDomainNameSet(void) const;
```

Reply:
Added this!